activation/post.go

```diff
@@ -287,13 +289,9 @@ func (mgr *PostSetupManager) GenerateProof(challenge []byte) (*types.Post, *type
 	}
 	mgr.mu.Unlock()
 
-	prover, err := proving.NewProver(config.Config(mgr.cfg), mgr.lastOpts.DataDir, mgr.id.Bytes(), mgr.commitmentAtxId.Bytes())
-	if err != nil {
-		return nil, nil, fmt.Errorf("new prover: %w", err)
-	}
-
-	prover.SetLogger(mgr.logger)
-	proof, proofMetadata, err := prover.GenerateProof(challenge)
+	proof, proofMetadata, err := proving.Generate(context.TODO(), challenge, config.Config(mgr.cfg), mgr.logger,
```

Comment:
Do we want to replace this with a context that can timeout/be cancelled ?

Reply:
I passed in the context from the parent. I didn't bother to do so initially, because the implementation in this PR is only temporary until we replace it with an implementation in Rust, which is probably not able to handle contexts 🙂